input/pipewire/dump.go

```diff
@@ -0,0 +1,186 @@
+package pipewire
+
+import (
+	"encoding/json"
+	"os/exec"
+
+	"github.com/pkg/errors"
+)
+
+type pwObjects []pwObject
+
+func pwDump() (pwObjects, error) {
+	dumpOutput, err := exec.Command("pw-dump").Output()
+	if err != nil {
+		var execErr *exec.ExitError
+		if errors.As(err, &execErr) {
+			return nil, errors.Wrapf(err, "failed to run pw-dump: %s", execErr.Stderr)
+		}
+		return nil, errors.Wrap(err, "failed to run pw-dump")
+	}
+
+	var dump pwObjects
+	if err := json.Unmarshal(dumpOutput, &dump); err != nil {
+		return nil, errors.Wrap(err, "failed to parse pw-dump output")
+	}
+
+	return dump, nil
+}
+
+func (d pwObjects) Links() pwObjects {
+	return d.
+		Filter(func(o pwObject) bool { return o.Type == pwInterfaceLink })
+}
+
+func (d pwObjects) AudioSinks() pwObjects {
```

Comment:
Can we change this to AudioSinksAndStreams, and use

```go
func(o pwObject) bool {
	return o.Info.Props.MediaClass == pwAudioSink || o.Info.Props.MediaClass == pwStreamOutputAudio
},
```
for the media class filter?

`pwStreamOutputAudio` is `"Stream/Output/Audio"`